airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_options/interpolated_request_input_provider.py

```diff
@@ -2,37 +2,38 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
+from dataclasses import InitVar, dataclass, field
 from typing import Any, Mapping, Optional, Union
 
 from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.types import Config, StreamSlice, StreamState
 
 
+@dataclass
 class InterpolatedRequestInputProvider:
     """
     Helper class that generically performs string interpolation on the provided dictionary or string input
     """
 
-    def __init__(
-        self, *, config: Config, request_inputs: Optional[Union[str, Mapping[str, str]]] = None, **options: Optional[Mapping[str, Any]]
-    ):
-        """
-        :param config: The user-provided configuration as specified by the source's spec
-        :param request_inputs: The dictionary to interpolate
-        :param options: Additional runtime parameters to be used for string interpolation
-        """
+    options: InitVar[Mapping[str, Any]]
+    request_inputs: Optional[Union[str, Mapping[str, str]]] = field(default=None)
```

Comment:
this class isn't exposed in the DSL, but I still abstained from modifying the name of the parameters. I chose to rename the method instead